wallet/wallet.go

```diff
@@ -834,30 +873,42 @@ func (sw *SingleAddressWallet) RedistributeV2(outputs int, amount, feePerByte ty
 			txn.SiacoinInputs = append(txn.SiacoinInputs, types.V2SiacoinInput{
 				Parent: sce.Move(),
 			})
-			sw.locked[sce.ID] = time.Now().Add(sw.cfg.ReservationDuration)
+			toLock = append(toLock, sce.ID)
 		}
 		txns = append(txns, txn)
 		toSign = append(toSign, toSignTxn)
 	}
+
+	if err := sw.lockUTXOs(toLock); err != nil {
+		return nil, nil, fmt.Errorf("failed to lock UTXOs: %w", err)
+	}
 	return
 }
 
 // ReleaseInputs is a helper function that releases the inputs of txn for use in
 // other transactions. It should only be called on transactions that are invalid
 // or will never be broadcast.
-func (sw *SingleAddressWallet) ReleaseInputs(txns []types.Transaction, v2txns []types.V2Transaction) {
+func (sw *SingleAddressWallet) ReleaseInputs(txns []types.Transaction, v2txns []types.V2Transaction) error {
 	sw.mu.Lock()
 	defer sw.mu.Unlock()
+	var released []types.SiacoinOutputID
 	for _, txn := range txns {
 		for _, in := range txn.SiacoinInputs {
 			delete(sw.locked, in.ParentID)
+			released = append(released, in.ParentID)
 		}
 	}
 	for _, txn := range v2txns {
 		for _, in := range txn.SiacoinInputs {
 			delete(sw.locked, in.Parent.ID)
+			released = append(released, in.Parent.ID)
 		}
 	}
+
+	if err := sw.store.ReleaseUTXOs(released); err != nil {
```

Comment:
I would collect the ids, update database and then delete them from memory. If `store.ReleaseUTXOs` errors out you'll have deleted them from `sw.locked` already.

Reply:
While I would generally agree that in most cases you'd do I/O first and then the in-memory update for consistency, it might be detrimental here.

If `ReleaseInputs` fails, nobody is going to retry it similar to how nobody retries closing a file if it returns an error. But we know that people still want to be able to spend that input again. So to avoid locking that input up I think releasing from the map might still make sense here.

Of course whether the error is temporary or not is a whole different question and depends on the implementation.